core/src/main/java/com/homeaway/streamingplatform/db/dao/impl/StreamDaoImpl.java

```diff
@@ -153,12 +165,16 @@ public void upsertStream(Stream stream) {
                 avroStream.setCreated(System.currentTimeMillis());
                 key = AvroStreamKey.newBuilder().setStreamName(avroStream.getName()).build();
             }
+
             verifyAndUpsertTopics(stream);
             kafkaProducer.log(key, avroStream);
             log.info("Stream upserted for {}", stream.getName());
         } catch (IllegalArgumentException e) {
             log.error("caught an illegal argument exception");
             throw new IllegalArgumentException(e);
```

Comment:
why catch and rethrow illegalargumentexception?

Reply:
not totally sure, removed it